convert_results.py

```diff
@@ -0,0 +1,130 @@
+outputs, use this script to extract the most interest information from your JSON
+reports and write them into an eye-friendly markdown file.
+
+  Typical usage example:
+
+  python convert_results.py --source=results/ --output=results/summaries/
+"""
+
+import argparse
+from collections import defaultdict
+import json
+from pathlib import Path
+
+import jinja2
+from tabulate import tabulate
+from tqdm.auto import tqdm
+
+
+def summary_table(tests):
+  """Creates a Markdown table summarizing passed tests for all tags.
+
+    Args:
+      tests: A list of tests, that have a 'result' entry, and possibly 'tags'.
+        The value of 'result' is 'pass' for a passing test.
+        If a test contains the key 'tags', it must be a list of strings.
+
+    Returns:
+      A string with a summary table in Markdown.
+  """
+  tags = defaultdict(lambda: defaultdict(int))
+  for test in tests:
+    # The 'All' tag is added to count all test results.
+    for tag in test.get('tags', []) + ['All']:
+      if test['result'] == 'pass':
+        tags[tag]['pass'] += 1
+      tags[tag]['total'] += 1
+
+  headers = ['Category', 'Passed', 'Total']
+  table = []
+  for tag, counts in tags.items():
+    table.append([tag, counts['pass'], counts['total']])
+  return tabulate(table, headers, tablefmt='github')
+
+
+def convert(result, template_file='template.md'):
+  """Converts the result file JSON content to a Markdown text.
+
+    Args:
+      result: A string with the content of a JSON file output by the test tool.
+      template_file: Markdown template for jinja.
+
+    Returns:
+      A string with the text of a Markdown file.
+  """
+  try:
+    result = json.loads(result)
+  except ValueError as _:
+    print('File was not valid JSON.')
+    return ''
+
+  for key in ['capabilities', 'device_under_test', 'tests']:
+    if key not in result:
+      print('JSON was missing the mandatory key {}.'.format(key))
+      return ''
+
+  capabilities = result['capabilities']
+  device_under_test = result['device_under_test']
+  tests = result['tests']
+  for test in tests:
+    if 'result' not in test:
+      test['result'] = 'fail'
+  failed_tests = [test for test in tests if test['result'] == 'fail']
+  summary = summary_table(tests)
+
+  loader = jinja2.FileSystemLoader(searchpath='./')
+  env = jinja2.Environment(loader=loader)
+  template = env.get_template(template_file)
+  return template.render(capabilities=capabilities,
+                         device_under_test=device_under_test,
+                         failed_tests=failed_tests,
+                         summary=summary)
+
+
+def main(args):
+  """Converts all JSON file from the source directory to Markdown."""
+  output_path = Path(args.output_dir)
+  if output_path.exists():
+    print('Directory', args.output_dir, 'exists, files may be overwritten.')
+  else:
+    output_path.mkdir(parents=True)
+  progress_bar = tqdm(Path(args.source_dir).glob('*.json'))
+  for path in progress_bar:
```

Comment:
I would do something different:

```
paths = list(Path(args.source_dir).glob('*.json'))
pb = tqdm(total=len(paths), unit='files')
for path in paths:
   pb.set_description()
   pb.update()
```
That would allows to not clamp the progress bar with the logic

Reply:
Calling `set_description()` with an empty argument list didn't seem to make any difference, so I removed it. The old `set_description()` would mess up the loading bar since the description length is not constant.